data/code/lib.c

```diff
@@ -589,3 +593,732 @@ void check_lap_ex()
+                                                            // Used to get X direction (forwards)
+                                                            // by flipping X and Z axes.
+
+    movement_vector[0] = (z_direction_vector[2] * -1.0) * x_z_speed;
+    movement_vector[2] = (z_direction_vector[0]) * x_z_speed;
+
+    // Set to be equal to the movement we want to perform in the game's eyes.
+    // Now, the game will not scale down the movement vector.
+    *velocity_frame =
+        ((movement_vector[0] * movement_vector[0]) + (movement_vector[1] * movement_vector[1]) +
+         (movement_vector[2] * movement_vector[2])) *
+        2.16 * *scale;
+
+    write_movement_vector(this, movement_vector[0], movement_vector[1], movement_vector[2]);
+}
+
+void decrement_mini_turbo_timer(char* const this)
+{
+    short* mini_turbo_timer = (short*)(this + MINI_TURBO_TIMER_OFFSET);
+    if (get_boost_flag(this, MINI_TURBO_FLAGS, MINI_TURBO_BOOST_FLAG) != 0)
+    {
+        --*mini_turbo_timer;
+
+        if (*mini_turbo_timer == 0)
+        {
+            clear_boost_flag(this, MINI_TURBO_FLAGS, MINI_TURBO_CLEAR_HASH_1);
+        }
+    }
+}
+
+void decrement_unknown_boost(char* const this)
+{
+    char* unknown_decrement = (char*)(this + UNKNOWN_TIMER_OFFSET);
+
+    if (*unknown_decrement != '\0')
+    {
+        --*unknown_decrement;
+    }
+
+    if (get_boost_flag(this, MINI_TURBO_FLAGS, 0x4) != 0 && *unknown_decrement == '\0')
+    {
+        clear_boost_flag(this, MINI_TURBO_FLAGS, MINI_TURBO_CLEAR_HASH_2);
+    }
+}
+
+// This function essentially recreates KartStrat::DoDash()'s boost timer decrementation.
+// Uglier than older implementaiton (triple if indentation), but doesn't underflow.
+void decrement_boost_timers(char* const this)
+{
+    short* dash_timer = (short*)(this + GENERIC_DASH_TIMER_OFFSET);
+    char* unknown_decrement = (char*)(this + UNKNOWN_TIMER_OFFSET);
+
+    decrement_mini_turbo_timer(this);
+    decrement_unknown_boost(this);
+
+    if (get_boost_flag(this, GENERIC_DASH_FLAGS, GENERIC_DASH_BOOST_FLAG) != 0)
+    {
+        --*dash_timer;
+
+        if (*dash_timer == 0)
+        {
+            if (get_boost_flag(this, GENERIC_DASH_FLAGS, GENERIC_DASH_GO_BOOST_FLAG) != 0)
+            {
+                set_boost_flag(this, MINI_TURBO_FLAGS, 0x4);
+                *unknown_decrement = 0xf;
+            }
+            clear_boost_flag(this, GENERIC_DASH_FLAGS, GENERIC_DASH_CLEAR_HASH);
+            clear_boost_flag(this, MINI_TURBO_FLAGS, MINI_TURBO_CLEAR_HASH_3);
+        }
+    }
+}
+
+// Boosts are usually handled by DoSpeedCtrl. Replicates its functionality
+// while also adding own logic.
+void handle_boosts(char* const this)
+{
+    decrement_boost_timers(this);
+}
+
+// Slows XZ movement during bounce while not pressing left or right.
+float deaccelerate_speed(float last_momentum)
+{
+    float decceleration = 0.004;
```

Comment:
As I may or may have not mentioned the other day, it'd be nice to mark as `const` variables and parameters that are not expected to change. It makes the code clearer, and is less error prone.

Here this would be:

```diff
- float deaccelerate_speed(float last_momentum)
+ float deaccelerate_speed(const float last_momentum)
```
```diff
-     float decceleration = 0.004;
+     const float decceleration = 0.004f;
```

Reply:
I've gone through and marked a few of these as `const`, including the `decceleration` variable suggested.